crates/typst-library/src/text/quote.rs

```diff
@@ -0,0 +1,136 @@
+/// ]
+/// #quote(
+///  block: true,
+///  author: [from the Henry Cary literal translation of 1897]
+/// )[
+///   ... I seem, then, in just this little thing to be wiser than this man at
+///   any rate, that what I do not know I do not think I know either.
+/// ]
+/// ```
+#[elem(Show)]
+pub struct QuoteElem {
+    /// Whether this is a block quote.
+    ///
+    /// ```example
+    /// #quote(author: [René Descartes])[cogito, ergo sum]
+    ///
+    /// #set quote(block: true)
+    /// #quote(author: [JFK])[Ich bin ein Berliner.]
+    /// ```
+    block: bool,
+
+    /// Whether double quotes should be added around this quote.
+    ///
+    /// - `{true}`: Wrap this quote in double quotes.
+    /// - `{false}`: Do not wrap this quote in double quotes.
+    /// - `{auto}`: Infer whether to wrap this quote in double quotes based on
+    ///   the `block` property. If `block` is `{false}` double quotes are
+    ///   auomatically added.
+    ///
+    /// ```example
+    /// #set text(lang: "de")
+    /// #quote[Ich bin ein Berliner.]
+    ///
+    /// #set smartquote(quotes: "«»")
+    /// #set quote(quotes: true)
+    /// #quote(block: true)[
+    ///   ... ἔοικα γοῦν τούτου γε σμικρῷ τινι αὐτῷ τούτῳ σοφώτερος εἶναι, ὅτι
+    ///   ἃ μὴ οἶδα οὐδὲ οἴομαι εἰδέναι.
+    /// ]
+    /// ```
+    quotes: Smart<bool>,
+
+    /// The source url to this quote.
+    ///
+    /// ```example
+    /// #show link: set text(blue)
+    /// #quote(source: "https://google.com")[cogito, ergo sum]
+    /// ```
+    source: Option<EcoString>,
+
+    /// The author of this quote.
+    ///
+    /// ```example
+    /// #quote(author: [René Descartes])[cogito, ergo sum] is the author. \
+    /// #quote[cogito, ergo sum] --- _Unknown_
+    /// ```
+    author: Option<Content>,
+
+    /// The quote.
+    #[required]
+    body: Content,
+}
+
+impl Show for QuoteElem {
+    fn show(&self, _: &mut Vt, styles: StyleChain) -> SourceResult<Content> {
+        let mut realized = self.body();
+        let author = self.author(styles);
+        let block = self.block(styles);
+
+        if self.quotes(styles).as_custom().is_some_and(|q| q) || !block {
+            let quote = SmartquoteElem::new().with_double(true).pack();
+            let weak_h = HElem::new(Spacing::Rel(Rel::zero())).with_weak(true).pack();
+
+            realized = Content::sequence([
+                quote.clone(),
+                weak_h.clone(),
+                realized,
+                weak_h,
+                quote,
+            ]);
+        }
+
+        if block {
+            realized = BlockElem::new().with_body(Some(realized)).pack();
+
+            if let Some(author) = author {
+                realized += Content::sequence([
+                    TextElem::packed('—'),
+                    SpaceElem::new().pack(),
+                    author,
+                ])
+                .aligned(Align::END);
+            }
+
+            let pad: Rel = Em::new(1.0).into();
```

Comment:
it doesn't seem ideal that this is hard-coded. somebody will want to customize it.

Reply:
Would using `Finalize` and setting it there be appropriate? That should enable `show quote: set pad(x: ...)` to change it, if I understood the usage in `figure.rs` right.

Reply:
Finalize is a bit wonky, but it would probably work. Should be documented though then.